tests/data_properties_v3_conversion/genius_zone_data_state_test.py

```diff
@@ -0,0 +1,248 @@
+        "Check that the temperature is not set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.OnOffTimer,
+            ZONE_TYPE.ControlOnOffPID,
+            ZONE_TYPE.Surrogate
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertFalse('temperature' in genius_zone.data)
+
+    def test_when_iType_should_set_setpoint_state_setpoint_set_correctly(self):
+        "Check that the setpoint is set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.ControlSP,
+            ZONE_TYPE.TPI
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertEqual(genius_zone.data['setpoint'], setpoint)
+
+    def test_when_iType_should_not_set_setpoint_state_setpoint_not_set(self):
+        "Check that the setpoint is not set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.Manager,
+            ZONE_TYPE.ControlOnOffPID,
+            ZONE_TYPE.Surrogate
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertFalse('setpoint' in genius_zone.data)
+
+    def test_when_iType_OnOffTimer_fSP_not_zero_setpoint_state_setpoint_set_true(self):
+        """Check that the setpoint is set to true when iType is OnOffTimer
+
+        and fSP is not zero"""
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        self.raw_json["iType"] = ZONE_TYPE.OnOffTimer
+        genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+        self.assertTrue(genius_zone.data['setpoint'])
+
+    def test_when_iType_OnOffTimer_fSP_zero_setpoint_state_setpoint_set_false(self):
+        """Check that the setpoint is set to false when iType is OnOffTimer
+
+        and fSP is zero"""
+
+        temperature = 20.0
+        setpoint = 0.0
+
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
```

Comment:
Why not:
```python
self.raw_json["fPV"] = 20.0  # temperature
```

Reply:
Because I may have gotten a bit cut'n'paste happy!